angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js

```diff
@@ -103,47 +102,85 @@ define(['angular'], function (angular) {
       }
     };
 
-    $scope.changePref = function (userPreference) {
+    $scope.cycleUnits = function (){
 
-      if (userPreference === 'C') {
-        $scope.changeToC();
-        $scope.showMetric = true;
-      } else {
-        $scope.changeToF();
+      var userPreference = $scope.nextUnits;
```

Comment:
Lines 91 - 93, there's a loop to call cyaleUnits() until $scope.currentUnits and $scope.nextUnits into alignment with the user's preferences. 

The data comes in as Fahrenheit, so I'm just initializing the values to an order the code is expecting before I call that loop. If the user's preference is F, then we avoid that loop altogether, and I need that C in nextUnits to be populated so it will display.